docs/_snippets/framework/tutorials/external-widget.js

```diff
@@ -0,0 +1,212 @@
+		this._defineSchema();
+		this._defineConverters();
+		this._updateWidgetData();
+
+		this.editor.commands.add( 'external', new ExternalWidgetCommand( this.editor ) );
+	}
+
+	_intervalFetch() {
+		return setInterval( () => this._updateWidgetData(), 15000 ); // set time interval to 15s
+	}
+
+	async _updateWidgetData( externalUrl = RESOURCE_URL ) {
+		try {
+			const response = await fetch( externalUrl );
+			const data = await response.json();
+			const updateTime = new Date( data.closeTime );
+			const parsedData = '$' + Number( data.lastPrice ).toFixed( 2 ) + ' - ' + updateTime.toLocaleString();
+
+			this.externalDataValue = parsedData;
+
+			const rootElement = this.editor.model.document.getRoot();
+
+			for ( const { item } of this.editor.model.createRangeIn( rootElement ) ) {
+				if ( item.is( 'element', 'externalElement' ) ) {
+					this.editor.editing.reconvertItem( item );
+				}
+			}
+		} catch ( error ) {
+			console.error( error );
+		}
+	}
+
+	_defineSchema() {
+		const schema = this.editor.model.schema;
+
+		schema.register( 'externalElement', {
+			inheritAllFrom: '$inlineObject',
+			allowAttributes: [ 'data-resource-url' ]
+		} );
+	}
+
+	_defineConverters() {
+		const editor = this.editor;
+
+		editor.conversion.for( 'upcast' ).elementToElement( {
+			view: {
+				name: 'span',
+				attributes: [ 'data-resource-url' ]
+			},
+			model: ( viewElement, { writer } ) => {
+				const externalUrl = viewElement.getAttribute( 'data-resource-url' );
+
+				return writer.createElement( 'externalElement', {
+					'data-resource-url': externalUrl
+				} );
+			}
+		} );
+
+		editor.conversion.for( 'dataDowncast' ).elementToElement( {
+			model: 'externalElement',
+			view: ( modelElement, { writer } ) => {
+				return writer.createEmptyElement( 'span', {
+					'data-resource-url': modelElement.getAttribute( 'data-resource-url' )
+				} );
+			}
+		} );
+
+		editor.conversion.for( 'editingDowncast' ).elementToElement( {
+			model: 'externalElement',
+			view: ( modelElement, { writer } ) => {
+				const externalValueToShow = this.externalDataValue;
+
+				const externalDataPreviewElement = writer.createRawElement( 'span', null, function( domElement ) {
+					domElement.textContent = externalValueToShow || 'Fetching data...';
+
+					if ( externalValueToShow ) {
+						domElement.classList.add( 'external-widget-bounce' );
+						setTimeout( () => domElement.classList.remove( 'external-widget-bounce' ), 1100 );
+					}
+				} );
+
+				const externalWidgetContainer = writer.createContainerElement( 'span', null, externalDataPreviewElement );
+
+				return toWidget( externalWidgetContainer, writer, {
+					label: 'External widget'
+				} );
+			}
+		} );
+	}
+}
+
+ClassicEditor
+	.create( document.querySelector( '#snippet-external-widget' ), {
+		plugins: [ Essentials, Paragraph, Heading, List, Bold, Italic, ExternalWidget ],
+		toolbar: [ 'heading', 'bold', 'italic', 'numberedList', 'bulletedList', '|', 'external', '|', 'undo', 'redo' ]
```

Comment:
Let's have `'external'` at the beginning of the toolbar followed by a separator (discoverability).

Reply:
Done aa8caf0